docs/create-analystics-event.md

```diff
@@ -0,0 +1,143 @@
+      event: MetaMetricsEventName.YourNewEventName,
+      category: MetaMetricsEventCategory.YourCategory,
+      properties: {
+        // Add properties that provide context about the event
+        property_one: 'value_one',
+        property_two: 'value_two',
+      },
+      sensitiveProperties: {
+        // Add properties that might contain sensitive data
+        // These will be automatically anonymized
+        wallet_address: '0x...',
+      },
+    });
+  };
+
+  return <button onClick={handleAction}>Perform Action</button>;
+}
+```
+
+#### 2.3 E2E Testing
+
+For implementing analytics event mocks in E2E tests using `testSpecificMock`, please refer to recording in notion of [Extension Codebase - E2E tests (mocking)](https://www.notion.so/metamask-consensys/f649ecf027cc41db858d3a3574fe3a99?v=da4af5e48f0e44d7b4536c4e2e911f36).
+
+Example E2E test implementation:
+
+```typescript
+// Mock the analytics endpoint
+async function mockAnalyticsEndpoint(mockServer) {
+  return await mockServer
+    .forPost('https://api.segment.io/v1/batch')
+    .withJsonBodyIncluding({
+      batch: [
+        {
+          type: 'track',
+          event: MetaMetricsEventName.YourNewEventName,
+          properties: {
+            property_one: 'value_one',
+            property_two: 'value_two',
+          },
+        },
+      ],
+    })
+    .thenCallback(() => ({ statusCode: 200 }));
+}
+
+describe('YourNewEventName Analytics', () => {
+  it('should track the event with correct properties', async () => {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        testSpecificMock: async (mockServer) => [
+          await mockAnalyticsEndpoint(mockServer),
+        ],
+      },
+      async ({ driver, mockedEndpoints }) => {
+        // Test implementation
+        const events = await getEventPayloads(driver, mockedEndpoints);
+        assert.equal(events[0].properties.property_one, 'value_one');
+        assert.equal(events[0].properties.property_two, 'value_two');
+      },
+    );
+  });
+});
+```
+
+#### 2.4. Validate the Event
+
+To validate your analytics event implementation:
+
+1. Run the extension locally in development mode
+2. Open Chrome DevTools
+3. Navigate to the Network tab
+4. Perform the action that should trigger your analytics event
+5. Look for a `batch` request in the Network tab
+6. Inspect the request payload to verify:
+   - The event name matches your implementation
+   - All expected properties are present and correctly formatted
+   - Property values are being captured as intended
+
+![Analytics Event Request Process](./images/create-analystics-event-request.png)
+
+#### 2.5. Opt-out Handling
+
+The tracking system automatically respects user privacy settings:
+
+- Events are only tracked if the user has opted into MetaMetrics
+- Sensitive data is automatically anonymized
```

Comment:
S: explain how this is anonymized. For instance on mobile we have:
- normal event: one event with all props and user id
- anonymous event: two events:
  - one normal event with only non sensitive props and user id (you know who but not what)
  - one anonymous event with sensitive props and an anonymous id (you know what but not who)

Reply:
Good idea, added in d5727a2b7bae39d5613c511d4f07207c26e37c0d